src/aleph_client/commands/utils.py

```diff
@@ -200,3 +206,111 @@ def is_environment_interactive() -> bool:
             not os.environ.get("DEBIAN_NONINTERACTIVE") == "noninteractive",
         )
     )
+
+
+from aleph_client.commands.node import NodeInfo, _fetch_nodes, _escape_and_normalize, _remove_ansi_escape, _format_score
+
+
+class ProgressTable:
+    def __init__(self, progress, table):
+        self.progress = progress
+        self.table = table
+
+    def __rich_console__(self, console, options):
+        yield self.progress
+        yield self.table
+
+
+async def fetch_crn_info():
+    node_info = await _fetch_nodes()
+    table = Table(title="Compute Node Information")
+    table.add_column("Score", style="green", no_wrap=True, justify="center")
+    table.add_column("Name", style="#029AFF", justify="left")
+    table.add_column("CPU", style="green", justify="left")
+    table.add_column("RAM", style="green", justify="left")
+    table.add_column("HDD", style="green", justify="left")
+    table.add_column("Version", style="green", justify="center")
+    table.add_column("Reward Address", style="green", justify="center")
+    table.add_column("Address", style="green", justify="center")
+
+    console = Console()
+    progress = Progress()
+    task = progress.add_task("[green]Fetching node info... It might take some time", total=len(node_info.nodes))
+
+    progress_table = ProgressTable(progress, table)
+    item_hashes = []
+
+    queue = asyncio.Queue()
+
+    async with aiohttp.ClientSession() as session:
+        with Live(progress_table, console=console, refresh_per_second=2):
+            fetch_task = asyncio.create_task(fetch_data(session, node_info, queue, progress, task, item_hashes))
+            update_task = asyncio.create_task(update_table(queue, table))
+            await asyncio.gather(fetch_task, update_task)
+
+    return item_hashes
+
+async def fetch_data(session: aiohttp.ClientSession, node_info: NodeInfo, queue: asyncio.Queue, progress: ProgressTable, task: list, item_hashes: list):
+    tasks = [fetch_and_queue(session, node, queue, progress, task, item_hashes) for node in node_info.nodes]
+    await asyncio.gather(*tasks)
+    await queue.put(None)
```

Comment:
What about https://docs.python.org/3/library/asyncio-queue.html#asyncio.Queue.task_done ?

Reply:
In my mind, it is not necessary to use this function because `asyncio.gather` already handles task synchronization and completion effectively.
Adding these functions would maybe increase complexity without providing a lot of benefit.